File: capacity_club_mobile_app/lib/features/home/domain/entity/collective-training-session.entity.dart
class CollectiveTrainingSession {
  final String collective_training_session_id;
  final String title;
  final DateTime training_date;
  final String start_hours;
  final String end_hours;
  final int nb_place;
  final int nb_subscription;
  //workout: Workout;

  const CollectiveTrainingSession(
      {required this.collective_training_session_id,
      required this.title,
      required this.training_date,
      required this.start_hours,
      required this.end_hours,
      required this.nb_place,
      required this.nb_subscription});

  factory CollectiveTrainingSession.empty() {
    return CollectiveTrainingSession(
        collective_training_session_id: '',
        title: '',
        training_date: DateTime.now(),
        start_hours: '',
        end_hours: '',
        nb_place: 0,
        nb_subscription: 0);
  }
}

File: capacity-club-api/src/domain-modules/collective-training-session/model/entity/collective-training-session.entity.ts
import { BaseEntity } from '@common/model';
import { Workout } from 'domain-modules/workout/model';
import { PrimaryColumn, Column, JoinColumn, OneToOne, Entity } from 'typeorm';

@Entity()
export class CollectiveTrainingSession extends BaseEntity {
  @PrimaryColumn('varchar', { length: 26 })
  collective_training_session_id: string;
  @Column({ nullable: false })
  title: string;
  @Column({ nullable: false })
  training_date: Date;
  @Column({ nullable: false })
  start_hours: string;
  @Column({ nullable: false })
  end_hours: string;
  @Column({ nullable: false })
  nb_place: number;

  @OneToOne(() => Workout)
  @JoinColumn({ name: 'workout_id', referencedColumnName: 'workout_id' })
  workout: Workout;
}
